### PR TITLE
[DataGrid] Introduce `minWidth` in `ColDef` for columns with fluid width

### LINKED ISSUE #1850: [DataGrid] Introduces `minWidth` in `ColDef` for columns with fluid width
Based on these discussions:
https://github.com/mui-org/material-ui-x/issues/1561#issuecomment-832753596
https://github.com/mui-org/material-ui-x/pull/1816#issuecomment-855771150

## Summary 💡

Have the ability to set `minWidth` for columns.

## Examples 🌈

A possible solution can be the following (adding optional `minWidth` for fluid columns):

```tsx
<DataGrid
  columns={[
    {
      field: 'id',
      flex: 1,
      minWidth: 50,
    },
    {
      field: 'username',
      width: 200,
    },
    {
      field: 'age',
      flex: 0.3,
      minWidth: 80,
    },
  ]}
  rows={{
    id: 1,
    username: '@MaterialUI',
    age: 20,
  }}
 />
```

## Motivation 🔦

In order for the fluid column width feature to be consistent, we need to introduce a way to set the `minWidth` of a fluid column.

## Benchmark

- https://ag-grid.com/javascript-grid/column-sizing/#column-flex

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Closes #1850

I have also reverted #1816 as said here https://github.com/mui-org/material-ui-x/issues/1850#issuecomment-856154201

Early review is definitely welcomed!

TODO:
- [ ] Add documentation: https://deploy-preview-1954--material-ui-x.netlify.app/components/data-grid/columns/
- [ ] Tests?

### CODE REVIEW

File: packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx [hunk 1]
```diff
@@ -197,7 +198,7 @@ export function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
       data-field={column.field}
       style={{
         width,
-        minWidth: width,
+        minWidth: minWidth !== undefined ? minWidth : width,
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx:201]: This min-width on the element has nothing to do with the min-width of #1850. See the benchmark link to better capture what the problem is.

**Review by oliviertassinari — CHANGES_REQUESTED** [review]

**Comment by DanailH** [comment]
@ZeeshanTamboli are you planning to work on this PR. I'm asking because there hasn't been much activity in the last 10 days and I am planning to pick it up. Just want to make sure we are not doing double work here 😉

**Comment by ZeeshanTamboli** [comment]
> @ZeeshanTamboli are you planning to work on this PR. I'm asking because there hasn't been much activity in the last 10 days and I am planning to pick it up. Just want to make sure we are not doing double work here 😉

@DanailH You can take this up, thanks. I have been aiming this PR but got some other priorities lately so haven't found time to work on this. Since I am not sure when I can continue working on this or something else, you can pick this up.

**Comment by DanailH** [comment]
@ZeeshanTamboli thanks for the quick reply - I'll keep this one open for now and use it as a base - thanks for the effort tho ;)

**Comment by DanailH** [comment]
I'm closing this PR in favour of that one https://github.com/mui-org/material-ui-x/pull/2101

**Pull request closed without merging**